src/components/Communicator/Communicator.actions.js

```diff
@@ -291,3 +298,84 @@ export function updateDefaultBoardsIncluded(boardAlreadyIncludedData) {
     defaultBoardsIncluded: boardAlreadyIncludedData
   };
 }
+
+export function syncCommunicators(remoteCommunicators) {
+  const reconcileCommunicators = (local, remote) => {
+    if (local.lastEdited && remote.lastEdited) {
+      if (moment(local.lastEdited).isAfter(remote.lastEdited)) {
+        return local;
+      }
+      if (moment(local.lastEdited).isBefore(remote.lastEdited)) {
+        return remote;
+      }
+      if (moment(local.lastEdited).isSame(remote.lastEdited)) {
+        return remote;
+      }
+    }
+    return remote;
+  };
+  const getActiveCommunicator = getState => {
+    return getState().communicator.communicators.find(
+      c => c.id === getState().communicator.activeCommunicatorId
+    );
+  };
+
+  return async (dispatch, getState) => {
+    const localCommunicators = getState().communicator.communicators;
+    const updatedCommunicators = [...localCommunicators];
+
+    for (const remote of remoteCommunicators) {
+      const localIndex = localCommunicators.findIndex(
+        local => local.id === remote.id
+      );
+
+      if (localIndex !== -1) {
+        // If the communicator exists locally, reconcile the two
+        const reconciled = reconcileCommunicators(
+          localCommunicators[localIndex],
+          remote
+        );
+        if (reconciled === localCommunicators[localIndex]) {
+          // Local is more recent, update the server
+          try {
+            const res = await dispatch(
+              updateApiCommunicator(localCommunicators[localIndex])
+            );
+            updatedCommunicators[localIndex] = res;
+          } catch (e) {
+            console.error(e);
+          }
+        } else {
+          updatedCommunicators[localIndex] = reconciled;
+        }
+      } else {
+        // If the communicator does not exist locally, add it
+        updatedCommunicators.push(remote);
+      }
+    }
+
+    const activeCommunicatorId = getActiveCommunicator(getState).id ?? null;
+    const lastRemoteSavedCommunicatorId = remoteCommunicators[0].id ?? null; //The last communicator saved on the server
+    const needToChangeActiveCommunicator =
+      activeCommunicatorId !== lastRemoteSavedCommunicatorId &&
+      updatedCommunicators.length &&
+      updatedCommunicators.findIndex(
+        communicator => communicator.id === lastRemoteSavedCommunicatorId
+      ) !== -1;
```

Comment:
also we could check if `lastRemoteSavedCommunicatorId` is not null.

Reply:
done